<launch>
    <arg name="namespace_prefix" default="easy_handeye_table" />
    <arg name="robot_ip" doc="The IP address of the UR3 robot" default="172.16.11.3" />

    <arg name="marker_size" doc="Size of the ArUco marker used, in meters" />
    <arg name="marker_id" doc="The ID of the ArUco marker used" />

    <!-- start the robot -->
    <include file="$(find ur_robot_driver)/launch/ur3_bringup.launch">
        <arg name="limited" value="false" />
        <arg name="robot_ip" value="$(arg robot_ip)" />
    </include>

    <include file="$(find ur3_moveit_config)/launch/ur3_moveit_planning_execution.launch">
    </include>
    
    <!--include file="$(find kinect2_bridge)/launch/kinect2_bridge.launch">
        <param name="fps_limit" type="double" value="120.0" />
    </include-->

    <include file="$(find azure_kinect_ros_driver)/launch/driver.launch">
        <!-- TODO: Add arguments here -->
        <!--param name="depth_enabled" type="bool" value="false" /-->
        <param name="point_cloud" type="bool" value="false" />
        <param name="fps" type="int" value="30" />
    </include>

    <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
        <param name="marker_size" type="double" value="6" />
        <param name="max_new_marker_error" type="double" value="0.02" />
        <param name="max_track_error" type="double" value="0.02" />
        <!--param name="output_frame" type="string" value="kinect2_rgb_optical_frame" /-->
        <param name="output_frame" type="string" value="rgb_camera_link" />

        <!--
        <remap from="camera_image" to="/kinect2/hd/image_color_rect" />
        <remap from="camera_info" to="/kinect2/hd/camera_info" />
        -->
        <remap from="camera_image" to="/rgb/image_raw" />
        <remap from="camera_info" to="/rgb/camera_info" />
    </node>
        
    <!-- start easy_handeye -->
    <include file="$(find easy_handeye)/launch/calibrate.launch" >
        <arg name="namespace_prefix" value="$(arg namespace_prefix)" />
        <arg name="eye_on_hand" value="false" />

        <!--arg name="tracking_base_frame" value="kinect2_rgb_optical_frame" /-->
        <arg name="tracking_base_frame" value="rgb_camera_link" />
        <arg name="tracking_marker_frame" value="ar_marker_2" />
        <arg name="robot_base_frame" value="base_link" />
        <arg name="robot_effector_frame" value="ee_link" />

        <arg name="freehand_robot_movement" value="false" />
        <arg name="robot_velocity_scaling" value="0.5" />
        <arg name="robot_acceleration_scaling" value="0.2" />

        <arg name="start_rviz" value="true" />
        <arg name="start_rqt" value="false" />

    </include>

</launch>
